evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -52,9 +51,6 @@ def fit(self, X, y=None):
         if (X.dtypes == bool).all():
             X = X.astype('category')
 
-        # Convert None to np.nan, since None cannot be properly handled
```

Comment:
I think since we do conversion to Woodwork, this isn't necessary anymore as WW handles this :)